src/map/utils/fishingutils.cpp

```diff
@@ -587,7 +599,7 @@ namespace fishingutils
                         CItem* PFish = PChar->UContainer->GetItem(0);
                         uint8 rankSkill = PChar->RealSkills.rank[SKILL_FISHING];
                         uint16 rankSkillCap = (rankSkill + 1) * 100;
-                        int32 charSkill = PChar->WorkingSkills.skill[SKILL_FISHING];
+                        int32 charSkill = PChar->RealSkills.skill[SKILL_FISHING];
```

Comment:
This is something that confused me too, I looked over the crafting cpp and it followed that same format as this fishing one, in that it used RealSkills when first querying your player skill level and WorkingSkills when looking if it was to do the skillups and that obviously works in that cpp. On my server though, after fishing skill level 4 all skillups stopped. I traced this to the fact that on line 641:

`                            if (((charSkill / 10) < fishMaxSkill) && ((charSkill / 10) < (rankSkillCap / 10)))`

charSkill, which was defined with WorkinSkills had a value of 128, so the code in the if statement would never run seeing that the player is supposedly above the skill cap (10), changing this to RealSkills fixed this and pulled the correct value of 40 for that player at the time and so allowed more skillups.

I am not sure where WorkingSkills is set and where it got that value of 128 from? As in, which skill is the working skill. But I guess this must be defined somewhere as the crafting cpp knows which skill it is currently working with. sorry if my change isn't the proper methodology for the function! muddling my way through tbh